src/System.Management.Automation/engine/DataStoreAdapterProvider.cs

```diff
@@ -257,7 +257,7 @@ public override string ToString()
         /// <returns>
         /// Read-only collection of strings representing the path separators characters for this provider.
         /// </returns>
-        public ReadOnlyCollection<string> PathSeparator { get; internal set; } = new ReadOnlyCollection<string>(new List<string>());
+        public ReadOnlyCollection<string> PathSeparator { get; set; } = new ReadOnlyCollection<string>(new List<string>());
```

Comment:
Why we need public set?

Reply:
Because the `WSMan` provider is in different namespace, thus the `set` property cannot be internal in order to be used there

Reply:
Ah, I forgot the S.M.A.dll is not internal visible to `Microsoft.WSMan.Management`.
We need to figure out a different way to do this then, because this property shouldn't have a public setter.